### ISSUE TITLE
Automatically import new functions from Function App if API management integration is enabled

### ISSUE THREAD

**Issue by praneetloke** [Azure/azure-functions-host#9552]
#### What problem would the feature you're requesting solve? Please describe.
When I add new functions to my function app, I have to manually import the newly-added functions as APIs in my API management service. This is the case even when I enable new HTTP methods for existing functions. There is no simple method to automate this right now, especially if you do not have an OpenAPI spec.

#### Describe the solution you'd like
Since the Function App is "aware" of the integration with API management service, it would be nice if modifications to the functions in the Function App are automatically synced with the API management service, therefore removing the need to constantly update the imported APIs.

#### Describe alternatives you've considered
I've looked at https://techcommunity.microsoft.com/t5/azure-paas-blog/import-azure-function-app-to-azure-api-management/ba-p/2594810. However, that requires maintaining an OpenAPI spec exported from the API management service which doesn't fit very well with the Function App programming model. Admittedly, it is better than having to manually do the import every time in the Portal. So while it can be considered a workaround it is still not a straightforward to keep the two in-sync.

#### Additional context
N/A

**Comment by kevin-boutin** [comment]
I agree with this issue. It is an absolute bear to automate the updates to APIM especially if creating a new API and not just an operation. As part of the function app deployment process, Azure should automatically update APIM if the function is integrated with an APIM.
